### PR TITLE
onlyoffice-bin_latest: 7.5.1 -> 8.0.0

### PR DESCRIPTION
**Pull request by fin444**
## Description of changes

[Changelog](https://github.com/ONLYOFFICE/DesktopEditors/blob/master/CHANGELOG.md#800)

**ONLYOFFICE is still broken on wlroots.** Thus, only the `_latest` package is being updated to preserve compatibility.

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [X] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by fin444** [comment]
The way the split-package setup was handled meant that it needed to be renamed every update, OfBorg interpreted this as a new package and it tripped the new test for `pkgs/by-name`. It seemed to me the best way to handle this was just to move ONLYOFFICE over to the new scheme. I think I did it correctly?

**Comment by infinisil** [comment]
I believe this was a false positive in the `pkgs/by-name` check (notably that's not ofborg). There's already a PR that would fix it: https://github.com/NixOS/nixpkgs/pull/285089

So there's no need to migrate this to `pkgs/by-name`. And indeed I wouldn't recommend it especially because of the complexity with multiple package versions, which `pkgs/by-name` can't deal with yet.

**Comment by infinisil** [comment]
Ohh sorry you're right, this is not a case where that PR would've helped. Looking at the first commit in this PR (https://github.com/NixOS/nixpkgs/pull/285888/commits/42a8cabeeb8a8270203a5d831c529f4258632dc5), for all CI can see, `onlyoffice-bin_8_0` is indeed a new package using `callPackage`, therefore it would have to go into `pkgs/by-name`.

On one hand, if you have completely separate expressions, migration to `pkgs/by-name` sounds good. However in this case I do see the potential to refactor these to have more shared code, which then wouldn't work the same in `pkgs/by-name`.

I'd say that's up to the maintainers, especially also the decision of whether it makes sense to keep pinned old versions around.

Something generally important though: Don't remove top-level attributes without adding at least a `throw` alias in `aliases.nix` to say why it got removed.

**Comment by fin444** [comment]
ONLYOFFICE versions past 7.2 are completely inoperable on wlroots (and they don't seem interested in fixing it), which is why the package was split in the first place. `onlyoffice-bin` is kept as-is for compatibility, while `onlyoffice-bin_latest` is updated. Over time, I would expect the two to continue to diverge, so I'm not sure combining them would be good in the long term.

But you are correct, I have added a throw alias.

**Comment by infinisil** [comment]
> Over time, I would expect the two to continue to diverge, so I'm not sure combining them would be good in the long term

Sounds good to keep them separate then!

File: pkgs/top-level/aliases.nix [hunk 1]
```diff
@@ -791,6 +791,7 @@ mapAliases ({
   openvpn_24 = throw "openvpn_24 has been removed, because it went EOL. 2.5.x or newer is still available"; # Added 2023-01-23
   orchis = orchis-theme; # Added 2021-06-09
   oni2 = throw "oni2 was removed, because it is unmaintained and was abandoned years ago."; #Added 2024-01-15
+  onlyoffice-bin_7_5 = throw "onlyoffice-bin_7_5 has been renamed to onlyoffice-bin_latest (and updated from 7.5.x)"; # Added 2024-02-03
```
Comment (infinisil) [inline pkgs/top-level/aliases.nix:794]: Alias for 7.2 is also needed

Comment (fin444) [reply pkgs/top-level/aliases.nix]: oh yes whoops

**Comment by fin444** [comment]
Also on the subject of `update.sh` - I copied it into both since that's how it was before. I'm not entirely sure how these work, but I don't really know if this was the correct move? They also just don't seem to have been working in the first place, even before the package split.

**Comment by fin444** [comment]
Ah, I was under the impression they were there for r-ryantm, not for the maintainers. They do still work, but looks like I broke `_latest`'s when updating to 7.4 because I put the `passthru` in the wrong spot. Have fixed that now.

I'm learning in this PR :laughing:

**Review by GTrunSec — APPROVED** [review]
LGTM

**Pull request merged**